zerver/lib/email_mirror_helpers.py

```diff
@@ -48,7 +49,12 @@ def get_email_gateway_message_string_from_address(address: str) -> str:
 
 
 def encode_email_address(stream: Stream, show_sender: bool = False) -> str:
-    return encode_email_address_helper(stream.name, stream.email_token, show_sender)
+    channel_email_address, ignored = ChannelEmailAddress.objects.get_or_create(
+        channel=stream,
+        creator=stream.creator,
+        sender=get_system_bot(settings.EMAIL_GATEWAY_BOT, stream.realm_id),
+    )
+    return encode_email_address_helper(stream.name, channel_email_address.email_token, show_sender)
```

Comment:
> If settings.EMAIL_GATEWAY_BOT is not enabled,

Internal bots are always present, right ? We don't suggest editing those settings in our docs.

Did you mean to say `settings.EMAIL_GATEWAY_PATTERN` ? If yes, then that'd be a separate PR to change the behavior of raising an error instead of returning an empty `""` string as the email.

Reply:
There might be some very old server that could be missing them, but I think that is probably a data problem with them, and the email gateway can't work without it, so maybe it's fine for this to be as-is?